### DISCUSSION TITLE
Are there docs on the .next folder generated after next build?

### DISCUSSION THREAD

**Discussion by gopinav** [vercel/next.js#24319] [Help]
Hello, I'm trying to understand the files and folders present in the .next folder but can't seem to find any docs about them.
Any help appreciated.

**Comment by akellbl4** [comment]
Next.js don't have docs on it. I think because it's just a build of your code.
`.next` contains two types of JS bundles: frontend javascript and backend javascript. Frontend JS files are chunks for client-side rendering and framework-related code. Backend chunks are javascript templates plus your data fetching functions for rendering pages. Also, it contains static files such as images, built styles, and so on.

**Reply by gopinav** [reply]
Thanks for the response. I'm trying to understand the differene between the 'static' and the 'server' folder after running 'next build'. Whats the role of each of them? If I'm running yarn start, I'm guessing the contents of the 'server' folder are required. When does 'static' folder come into picture? Thanks again.

**Reply by akellbl4** [reply]
`server` - files that will be run on the build stage for generation HTML pages. Also, they can cun for Incremental static regeneration, and of course, if you have server-side rendering.
`static` - frontend related files such as scripts, styles, images, etc. Those files that the browser can download to make pages work. Scripts contain framework-related code, vendors code (such as React), and code of the pages.

All of the files in `.next` are required if you run your app with `next start`. In case you use `next export` what you need is `out` folder.

**Reply by gopinav** [reply]
Thanks for the additional info. Starting to make a bit more sense now. I do have a few more questions now that I am taking a deeper look.
1. The HTML filesin `server/pages` are the ones that are sent back to the browser when user hits a route, right?
2. I have an SSR page users.js. I see `server/pages/users.js` and also `static/pages/users-hash.js`. Why do we have two files here? What does the code in each file represent?
3. The `users.json` file is the output of `getStaticProps` but I see `users.json` is sent to the browser even though `users.html` already has the data within the file. Would you happen to know why .json file from `server/pages/` is downloaded in the browser?
4. Am I wrong in assuming there is a bug in the pages listed after build? In the below screenshot, output reads (4/4) pages generated. However 500.html is not shown under the list of pages. Thank you for your time
![image](https://user-images.githubusercontent.com/13146172/115679277-07f39280-a370-11eb-9cbf-9cf006a95bea.png)

**Reply by akellbl4** [reply]
> 1. The HTML filesin server/pages are the ones that are sent back to the browser when user hits a route, right?

Right

> 2. I have an SSR page users.js. I see `server/pages/users.js` and also static/pages/users-hash.js. Why do we have two files here? What does the code in each file represent?
`server/pages/users.js` - uses for server rendering to HTML or for incremental regeneration on the server. Next.js runs these files and has HTML as output.

`static/pages/users-hash.js` – client-side chunk for the `/users` page contains all of the code to make your page dynamic. Also, when you come to a page via `next/link` Next.js actually doesn't render every next page on the server (and doesn't return static HTML) it gets `users.json` and renders the page fully on the client-side.
Try to go through your site and look at the developer console in the network tab and you'll see how Next.js is loading js chunks for the next page and JSON for rendering data on it.

> 3. The users.json file is the output of getStaticProps but I see users.json is sent to the browser even though users.html already has the data within the file. Would you happen to know why .json file from server/pages/ is downloaded in the browser?

Look at the previous answer :)

> 4. Am I wrong in assuming there is a bug in the pages listed after build? In the below screenshot, output reads 
(4/4) pages generated. However 500.html is not shown under the list of pages. Thank you for your time

Yep, I think we should see `/500` here.

Happy to help and explain it :)

**Reply by gopinav** [reply]
Thanks a lot for your detailed answer. It's a whole lot clearer now! I'll try to closely observe the network tab and inspect the files to understand further :)
